diff --git a/node_modules/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift b/node_modules/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift
index 284a0aa..80ce21e 100644
--- a/node_modules/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift
+++ b/node_modules/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift
@@ -4,15 +4,28 @@ import Foundation
 var subscribers = [ExpoAppDelegateSubscriberProtocol]()
 var reactDelegateHandlers = [ExpoReactDelegateHandler]()
 
+
+#if os(iOS) || os(tvOS)
+public typealias EMCResponder = UIResponder
+public typealias EMCApplicationDelegate = UIApplicationDelegate
+public typealias EMCWindow = UIWindow
+#elseif os(macOS)
+public typealias EMCResponder = NSResponder
+public typealias EMCApplicationDelegate = NSApplicationDelegate
+public typealias EMCWindow = NSWindow
+#endif
+
 /**
  Allows classes extending `ExpoAppDelegateSubscriber` to hook into project's app delegate
- by forwarding `UIApplicationDelegate` events to the subscribers.
+ by forwarding `UIApplicationDelegate`/`NSApplicationDelegate`  events to the subscribers.
 
- Keep functions and markers in sync with https://developer.apple.com/documentation/uikit/uiapplicationdelegate
+ Keep functions and markers in sync with:
+ - https://developer.apple.com/documentation/uikit/uiapplicationdelegate
+ - https://developer.apple.com/documentation/appkit/nsapplicationdelegate
  */
 @objc(EXExpoAppDelegate)
-open class ExpoAppDelegate: UIResponder, UIApplicationDelegate {
-  open var window: UIWindow?
+open class ExpoAppDelegate: EMCResponder, EMCApplicationDelegate {
+  open var window: EMCWindow?
 
   @objc
   public let reactDelegate = ExpoReactDelegate(handlers: reactDelegateHandlers)
@@ -321,6 +334,47 @@ open class ExpoAppDelegate: UIResponder, UIApplicationDelegate {
   }
 #endif
 
+#elseif os(macOS)
+  // MARK: - Initializing the App
+  public func applicationDidFinishLaunching(_ notification: Notification) {
+      subscribers.forEach { subscriber in
+          subscriber.applicationDidFinishLaunching?(notification)
+      }
+  }
+
+  public func applicationWillFinishLaunching(_ notification: Notification) {
+      let parsedSubscribers = subscribers.filter {
+          $0.responds(to: #selector(applicationWillFinishLaunching(_:)))
+      }
+
+      parsedSubscribers.forEach { subscriber in
+          subscriber.applicationWillFinishLaunching?(notification)
+      }
+  }
+
+  public func applicationDidBecomeActive(_ notification: Notification) {
+      subscribers.forEach { $0.applicationDidBecomeActive?(notification) }
+  }
+
+  public func applicationWillResignActive(_ notification: Notification) {
+      subscribers.forEach { $0.applicationWillResignActive?(notification) }
+  }
+
+  public func applicationWillTerminate(_ notification: Notification) {
+      subscribers.forEach { $0.applicationWillTerminate?(notification) }
+  }
+
+  @objc public func customizeRootView(_ rootView: NSView) {
+    subscribers.forEach { $0.customizeRootView?(rootView) }
+  }
+
+  public func application(_ application: NSApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+      subscribers.forEach { $0.application?(application, didRegisterForRemoteNotificationsWithDeviceToken: deviceToken) }
+  }
+
+  public func application(_ application: NSApplication, didFailToRegisterForRemoteNotificationsWithError error: any Error) {
+      subscribers.forEach { $0.application?(application, didFailToRegisterForRemoteNotificationsWithError: error) }
+  }
   #endif // os(iOS)
 
   // MARK: - Statics
