diff --git a/node_modules/react-native-macos/scripts/cocoapods/autolinking.rb b/node_modules/react-native-macos/scripts/cocoapods/autolinking.rb
index b2e5600..1ad96a1 100644
--- a/node_modules/react-native-macos/scripts/cocoapods/autolinking.rb
+++ b/node_modules/react-native-macos/scripts/cocoapods/autolinking.rb
@@ -15,7 +15,6 @@ require_relative '../react_native_pods.rb'
 # Parameters:
 # - config_command: the command to run to get the application's current config, e.g. ['npx', '@react-native-community/cli', 'config']
 def list_native_modules!(config_command)
-
   if !(config_command.is_a? Array and config_command.size > 0)
     Pod::UI.warn "Expected a list_native_modules! to be called with a config command", [
       "Unable to autolink if no config is provided for the current project."
@@ -50,6 +49,11 @@ def list_native_modules!(config_command)
     script_phases = package_config["scriptPhases"]
     configurations = package_config["configurations"]
 
+    if name == "react-native-macos"
+      next
+    end
+
+
     # Add a warning to the queue and continue to the next dependency if the podspec_path is nil/empty
     if podspec_path.nil? || podspec_path.empty?
       Pod::UI.warn("list_native_modules! skipped the react-native dependency '#{name}'. No podspec file was found.",
